src/domain/application/components/ApplicationContacts.tsx

```diff
@@ -76,33 +76,63 @@ const CustomerFields: React.FC<{
 
   const applicationType = getValues('applicationData.applicationType');
   const selectedContactType = watch(`applicationData.${customerType}.customer.type`);
+  const registryKey = watch(`applicationData.${customerType}.customer.registryKey`);
+  const registryKeyHidden = watch(`applicationData.${customerType}.customer.registryKeyHidden`);
   const registryKeyInputEnabled = isRegistryKeyInputEnabled(
     customerType,
     selectedContactType,
     applicationType,
   );
-  const isMounted = useRef(false);
+  const isMountedForSelectedContactType = useRef(false);
+  const isMountedForRegistryKey = useRef(false);
```

Comment:
Are both of these isMounted variables needed? Are there some problems if using only one in both useEffects?